network/src/commonJvmAndroid/kotlin/com/wire/kalium/network/HttpEngine.kt

```diff
@@ -37,67 +37,70 @@ import javax.net.ssl.SSLContext
 import javax.net.ssl.TrustManager
 import javax.net.ssl.X509TrustManager
 
+private object OkHttpSingleton {
+    private val sharedClient = OkHttpClient.Builder().apply {
+        pingInterval(WEBSOCKET_PING_INTERVAL_MILLIS, TimeUnit.MILLISECONDS)
+            .connectTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+            .readTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+            .writeTimeout(WEBSOCKET_TIMEOUT, TimeUnit.MILLISECONDS)
+    }.build()
+
+    fun createNew(block: OkHttpClient.Builder.() -> Unit): OkHttpClient {
+        return sharedClient.newBuilder().apply(block).build()
+    }
+}
+
 actual fun defaultHttpEngine(
     serverConfigDTOApiProxy: ServerConfigDTO.ApiProxy?,
     proxyCredentials: ProxyCredentialsDTO?,
     ignoreSSLCertificates: Boolean,
     certificatePinning: CertificatePinning
 ): HttpClientEngine = OkHttp.create {
+    OkHttpSingleton.createNew {
+        if (certificatePinning.isNotEmpty()) {
+            val certPinner = CertificatePinner.Builder().apply {
+                certificatePinning.forEach { (cert, hosts) ->
+                    hosts.forEach { host ->
+                        add(host, cert)
+                    }
+                }
+            }.build()
+            certificatePinner(certPinner)
+        }
 
-    val okHttpClient = OkHttpClient.Builder()
+        if (ignoreSSLCertificates) ignoreAllSSLErrors()
 
-    if (certificatePinning.isNotEmpty()) {
-        val certPinner = CertificatePinner.Builder().apply {
-            certificatePinning.forEach { (cert, hosts) ->
-                hosts.forEach { host ->
-                    add(host, cert)
+        // OkHttp doesn't support configuring ping intervals dynamically,
+        // so they must be set when creating the Engine
+        // See https://youtrack.jetbrains.com/issue/KTOR-4752
```

Comment:
Hasn't this comment lost its context? Shouldn't it be where the` pingInternval` is configured.